test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java

```diff
@@ -1086,4 +1090,48 @@ public void testLookupInKey() throws Exception{
         assertThat("overlapping ranges did not return intersection", hits5.guids, is(set(guid2)));
     }
 
+
+    @Test
+    public void addHighlighting() throws Exception {
+        GUID guid1 = new GUID("WS:11/1/2");
+        GUID guid2 = new GUID("WS:11/2/2");
+        GUID guid3 = new GUID("WS:11/3/2");
+//        try {
+            prepareTestMultiwordSearch(guid1, guid2, guid3);
+//        } catch(Exception e) {
+//            //TODO: remove items from each test after completion. Will error if run all sequence in order due to version conflict
+//        }
+        PostProcessing pp = new PostProcessing();
+        List<String> empty = new ArrayList<>();
+        
+        pp.objectHighlight = true;
+
+        List<kbasesearchengine.search.SortingRule> sorting = null;
+        AccessFilter accessFilter = AccessFilter.create().withAdmin(true);
+
+        //searchObjects
+        final Builder filter = MatchFilter.getBuilder();
+        filter.withNullableFullTextInAll("multiWordInSearchMethod1 multiWordInSearchMethod2");
+
+        FoundHits hits = indexStorage.searchObjects(empty, filter.build(),sorting, accessFilter, null, pp);
+        Map<String, List<String>> hitRes = hits.objects.get(0).getHighlight();
+
+        Map<String, List<String>> result1 = new HashMap<>();
+        result1.put("prop1", Arrays.asList("<em>multiWordInSearchMethod1</em> <em>multiWordInSearchMethod2</em>"));
+        assertThat("Incorrect highlighting", hitRes, is(result1) );
+
+        //searchIds is a wrapper around queryHits and does not r<em>WS:11/1/2</em>eturn object data and so will not be highlighted
```

Comment:
What's with the embedded guid...?

Reply:
accidentally copied and pasted that in there